extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -152,29 +154,52 @@ describe('Project: sqlproj content operations', function (): void {
 });
 
 describe('Project: round trip updates', function (): void {
-	before(async function () : Promise<void> {
+	before(async function (): Promise<void> {
 		await baselines.loadBaselines();
 	});
 
 	it('Should update SSDT project to work in ADS', async function (): Promise<void> {
-		await testUpdateInRoundTrip(baselines.SSDTProjectFileBaseline, baselines.SSDTProjectAfterUpdateBaseline);
+		const fileBeforeUpdate = baselines.SSDTProjectFileBaseline;
+		const fileAfterUpdate = isWindows ? baselines.SSDTProjectAfterUpdateBaselineWindows : baselines.SSDTProjectAfterUpdateBaseline;
+		await testUpdateInRoundTrip(fileBeforeUpdate, fileAfterUpdate, true, true);
+	});
+
+	it('Should update SSDT project with new system database references', async function (): Promise<void> {
+		const fileBeforeUpdate = isWindows ? baselines.SSDTUpdatedProjectBaselineWindows : baselines.SSDTUpdatedProjectBaseline;
+		const fileAfterUpdate = isWindows ? baselines.SSDTUpdatedProjectAfterSystemDbUpdateBaselineWindows : baselines.SSDTUpdatedProjectAfterSystemDbUpdateBaseline;
+		await testUpdateInRoundTrip(fileBeforeUpdate, fileAfterUpdate, false, true);
 	});
 
 	it('Should update SSDT project to work in ADS handling pre-exsiting targets', async function (): Promise<void> {
-		await testUpdateInRoundTrip(baselines.SSDTProjectBaselineWithCleanTarget, baselines.SSDTProjectBaselineWithCleanTargetAfterUpdate);
+		await testUpdateInRoundTrip(baselines.SSDTProjectBaselineWithCleanTarget, baselines.SSDTProjectBaselineWithCleanTargetAfterUpdate, true, false);
 	});
 });
 
-async function testUpdateInRoundTrip(fileBeforeupdate: string, fileAfterUpdate:string) : Promise<void> {
+async function testUpdateInRoundTrip(fileBeforeupdate: string, fileAfterUpdate: string, testTargets: boolean, testReferences: boolean): Promise<void> {
 	projFilePath = await testUtils.createTestSqlProjFile(fileBeforeupdate);
 	const project: Project = new Project(projFilePath);
 	await project.readProjFile();
 
-	await project.updateProjectForRoundTrip();
+	if (testTargets) {
+		await testUpdateTargetsImportsRoundTrip(project);
+	}
 
-	should(await exists(projFilePath + '_backup')).equal(true);	// backup file should be generated before the project is updated
-	should(project.importedTargets.length).equal(3);	// additional target added by updateProjectForRoundTrip method
+	if (testReferences) {
+		await testAddReferencesInRoundTrip(project);
+	}
 
 	let projFileText = (await fs.readFile(projFilePath)).toString();
 	should(projFileText).equal(fileAfterUpdate.trim());
 }
+
+async function testUpdateTargetsImportsRoundTrip(project: Project): Promise<void> {
+	await project.updateProjectForRoundTrip();
+	should(await exists(projFilePath + '_backup')).equal(true);	// backup file should be generated before the project is updated
+	should(project.importedTargets.length).equal(3);	// additional target added by updateProjectForRoundTrip method
```

Comment:
Can you also have it validate that the number of importedTargets is 2 _before_ the update; that way, we know for sure that it added a new one instead of already being at the correct goal state.